tests/rptest/tests/transactions_test.py

```diff
@@ -0,0 +1,235 @@
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from ducktape.utils.util import wait_until
+from rptest.clients.types import TopicSpec
+
+import time
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.rpk_consumer import RpkConsumer
+import confluent_kafka as ck
+from rptest.clients.kafka_cat import KafkaCat
+
+import subprocess
+
+
+class TransactionsTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=1, replication_factor=3),
+              TopicSpec(partition_count=1, replication_factor=3))
+
+    def __init__(self, test_context):
+        extra_rp_conf = {
+            "enable_idempotence": True,
+            "enable_transactions": True,
+            "transaction_coordinator_replication": 3,
+            "id_allocator_replication": 3,
+            "enable_leader_balancer": False,
+            "enable_auto_rebalance_on_node_add": False
+        }
+
+        super(TransactionsTest, self).__init__(test_context=test_context,
+                                               extra_rp_conf=extra_rp_conf)
+
+        self.input_t = self.topics[0]
+        self.output_t = self.topics[1]
+        self.max_records = 100
+
+    def generate_data(self, topic, num_records):
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+        })
+
+        for i in range(num_records):
+            producer.produce(topic.name, str(i), str(i), 0)
+
+        producer.flush()
+
+    def check_consumer(self, consumer, timeout_for_consume=2):
+        def consume_records():
+            for_one_consume = 10
+            self.records = consumer.consume(for_one_consume,
+                                            timeout_for_consume)
+            return (self.records != None) and (len(self.records) != 0)
+
+        wait_until(consume_records,
+                   timeout_sec=30,
+                   backoff_sec=2,
+                   err_msg="Can not consume data")
+
+    @cluster(num_nodes=3)
+    def simple_test(self):
+        self.generate_data(self.input_t, self.max_records)
+
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': '0',
+            'transaction.timeout.ms': 10000,
+        })
+
+        consumer1 = ck.Consumer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'group.id': "test",
+            'auto.offset.reset': 'earliest',
+            'enable.auto.commit': False,
+        })
+
+        producer.init_transactions()
+
+        consumer1.subscribe([self.input_t])
+
+        num_consumed_records = 0
+        while num_consumed_records != self.max_records:
+            self.check_consumer(consumer1)
```

Comment:
imagine that consume got broken, we read the same record twice and overshoot the condition such that  `num_consumed_records > self.max_records`. in this case we'll stuck for ever because `num_consumed_records != self.max_records` is always true. let's add an asert to prevent it:

```
while num_consumed_records != self.max_records:
    assert num_consumed_records < self.max_records
    self.check_consumer(consumer1)
```